centaur/src/main/scala/centaur/test/Test.scala

```diff
@@ -780,13 +780,39 @@ object Operations extends StrictLogging {
       }
 
       def checkArchived(): IO[Boolean] = for {
-        archiveStatus <- CentaurCromwellClient.archiveStatus(workflowId)
+        archiveStatus <- CentaurCromwellClient.archiveStatus(submittedWorkflow.id)
         isArchived <- validateMetadataArchiveStatus(archiveStatus)
       } yield isArchived
 
+      def validateMetadataSourceArchived(): IO[Boolean]  = for {
+        metadataSource <- CentaurCromwellClient.metadataWithId(submittedWorkflow.id).map(_.asFlat.stringifyValues.get("metadataSource"))
+        isMetadataSourceArchived <- {
+          if (metadataSource == Option(JsString("Archived"))) {
+            IO.pure(true)
+          } else if (metadataSource == Option(JsString("Unarchived"))) {
+            IO.pure(false)
+          } else {
+            throw CentaurTestException(
+              s"`Metadata` endpoint returned unknown value for `metadataSource`: $metadataSource",
+              workflowDefinition,
+              submittedWorkflow
+            )
+          }
+        }
+      } yield isMetadataSourceArchived
+
       def eventuallyArchived(): IO[Unit] = {
         checkArchived() flatMap {
-          case true => IO.pure(())
+          case true =>
+            validateMetadataSourceArchived() flatMap {
```

Comment:
TOL: Could you move this call inside the `checkArchived` for-comp to keep this part of the logic cleaner?

eg something like:
```
      def checkArchived(): IO[Boolean] = for {
        archiveStatus <- CentaurCromwellClient.archiveStatus(workflowId)
        isArchived <- validateMetadataArchiveStatus(archiveStatus)
        metadataSourceReturnsArchived <- validateMetadataSourceArchived()
      } yield isArchived && metadataSourceReturnsArchived
```